components/script/dom/servoparser/mod.rs

```diff
@@ -967,6 +965,10 @@ impl FetchResponseListener for ParserContext {
                 parser.parse_sync(CanGc::note());
             },
         }
+
+        if self.is_synthesized_document {
+            parser.document.mark_internal_content();
+        }
```

Comment:
I wonder if we should sanitize the input that created this document.
This change makes it so that synthesized documents can access `navigator.servo`.

From quickly glancing over the `match` block above, I can see two cases where user input could potentially 
end up unescaped in the document as part of some kind of error message (think log4j):
https://github.com/servo/servo/blob/d2afe00f7b6ad03488e08316f06f95f356f7edc5/components/script/dom/servoparser/mod.rs#L937-L950

Not sure if it's actually possible, but seems dangerous.

Since the input is added in a place where the tokenizer will be in the `Data` state, something like `xyz.replace("<", "&lt;"` is *probably* sufficient.

Reply:
I wonder if there's some way we could use the trusted types infrastructure here...